src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -314,20 +314,35 @@ NamedIntrinsic HWIntrinsicInfo::lookupId(Compiler*         comp,
         return NI_Throw_PlatformNotSupportedException;
     }
 
+    // Special case: For Vector64/128/256 we currently don't accelerate any of the methods when
```

Comment:
Just noting that one of the reasons these are "special case" is because `Vector128` is considered "ABI" stable with just `SSE` and `Vector256` is considered "ABI" stable with just `AVX`.

That is, `Vector256` corresponds to `__m256` at the ABI level and provided `AVX` is supported, then the handling of it when passed around as an HVA is expected to work.

However, for APIs called on the type, what can be accelerated or not depends on whether `AVX` or `AVX2` is supported. When its just `AVX`, then only `float/double` can really be accelerated. Integer types, like `int`, `byte`, `short`, or `ulong` require `AVX2` support for proper acceleration.

It's therefore easiest for the mapping information to have `Vector128` mapped to `SSE` and `Vector256` to be mapped to `AVX`. But for this import function to specially say `NI_Illegal` for any members exposed (like `Vector256.AndNot`) if the type is reporting `IsHardwareAccelerated == false`.